### ISSUE TITLE
Badblocks not detecting problems on a known to be failing USB memory stick. 

### ISSUE THREAD

**Issue by user-653** [tytso/e2fsprogs#77]
TLDR: Running `badblocks -sw /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:` on a USB memory stick that is known to corrupt data finishes without error.

I have a old 8GB Verbatim Store-n-Go USB that was corrupting data on a ext4 file system. Copying files to the STORE_N_GO with Gnomes file manager did not bring-up error messages in the GUI but copying from the STORE_N_GO did bring up warnings about IO errors while copying some files. Running e2fsck -c did not show problems.

From there I tried the following: 
Run  `badblocks -sw /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:` = No errors
Run  `badblocks -sw -t random /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:` = No errors
format with btrfs -m = scrubing warns about errors.

This made me believe the error only shown up on somewhat random access writes as done by a file system writing data to one area and metadata to a another in quick succession.

BUT running 
```
for i in {1..2}; do pv 8GB-bad\ \[4C9D3D1D\].img > /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:0 ; rhash --simple --uppercase  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:0 ;  rhash --simple --uppercase  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:0 ;  rhash --simple --uppercase  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0\:0 ; done
7.21GiB 0:34:47 [3.54MiB/s] [=========================================================================================================================================>] 100%            
B3D2829A  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
B3D2829A  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
B3D2829A  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
7.21GiB 0:34:47 [3.54MiB/s] [=========================================================================================================================================>] 100%            
3CA97365  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
3CA97365  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
3CA97365  /dev/disk/by-id/usb-Verbatim_STORE_N_GO_90006B6E8FEC8E73-0:0
```
Shows that sequential writes as I believe is done by badblocks should detect the error. The file 8GB-bad\ \[4C9D3D1D\].img was created by making a loop device the size of the STORE_N_GO and running `badblocks -sw -t random` on it to fill it with random data. The resulting CRC32 is 4C9D3D1D.

At this point I am at a loss on what information I need to provided for you to diagnose this false negative.

MY SYSTEM:
OS: Pop!_OS 20.04 LTS x86_64
Kernel: 5.11.0-7620-generic
e2fsprogs: 1.45.5-2ubuntu1
